runtime/src/lib.rs

```diff
@@ -414,45 +420,56 @@ mod validation_logic {
 				},
 			},
 			Categories::Shards(shards_script_info_struct) => {
-				// let format = shards_script_info_struct.format;
-				let requiring = &shards_script_info_struct.requiring;
-				// let implementing = &shards_script_info_struct.implementing;
+				let format = shards_script_info_struct.format;
+
+				// only support EDN for now
+				if format != ShardsFormat::Edn {
+					return false
+				}
 
-				// check that we reference all required traits
-				let all_required_trait_impls_found = requiring.iter().all(|shards_trait| {
+				// check trait exists and that we require proper references
+				let requiring = &shards_script_info_struct.requiring;
+				let require_check = requiring.iter().all(|trait_hash| {
 					// go thru all the things we reference, find shards scripts and check if they implement the trait
-					proto_references.iter().any(|proto_hash| {
+					if proto_references.iter().any(|proto_hash| {
 						if let Some(proto) = pallet_protos::Protos::<Runtime>::get(proto_hash) {
 							match proto.category {
 								Categories::Shards(shards_info) =>
-									shards_info.implementing.contains(shards_trait),
+									shards_info.implementing.contains(trait_hash),
 								_ => false,
 							}
 						} else {
 							false
 						}
-					})
+					}) {
+						if let Some(_) = pallet_protos::Traits::<Runtime>::get(trait_hash) {
```

Comment:
isn't this check redundant?

If one of the references has the `trait_hash` in its "implementing" field, obviously it exists, no?

Reply:
true